cmd/kube-apiserver/app/testing/testserver.go

```diff
@@ -408,7 +477,46 @@ func StartTestServer(t ktesting.TB, instanceOptions *TestServerInstanceOptions,
 	if err != nil {
 		return result, err
 	}
-	server, err := app.CreateServerChain(completed)
+
+	var server *apiserver.APIAggregator
+
+	if useTestAPI {
```

Comment:
Instead of a single fixed test API, I feel like it would be better to do it like in CRDs where we have a fixtures list: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/test/integration/fixtures/resources.go.

Tests can then selectively choose which fixture they would like added.